pandas/tests/arrays/test_array.py

```diff
@@ -0,0 +1,109 @@
+import decimal
+
+import numpy as np
+import pytest
+
+from pandas.core.dtypes.dtypes import registry
+
+import pandas as pd
+from pandas.api.extensions import register_extension_dtype
+from pandas.core.arrays import integer_array, period_array
+from pandas.tests.extension.decimal import (
+    DecimalArray, DecimalDtype, to_decimal)
+import pandas.util.testing as tm
+
+
+@pytest.mark.parametrize("data, dtype, expected", [
+    ([1, 2], None, np.array([1, 2])),
+    ([1, 2], object, np.array([1, 2], dtype=object)),
+    ([1, 2], 'float32', np.array([1., 2.0], dtype=np.dtype('float32'))),
+    ([pd.Period('2000', 'D'), pd.Period('2001', 'D')], 'Period[D]',
+     period_array(['2000', '2001'], freq='D')),
+    ([pd.Period('2000', 'D')], pd.PeriodDtype('D'),
+     period_array(['2000'], freq='D')),
+    ([1, 2], 'datetime64[ns]', np.array([1, 2], dtype='datetime64[ns]')),
+    (['a', 'b'], 'category', pd.Categorical(['a', 'b'])),
+    (['a', 'b'], pd.CategoricalDtype(None, ordered=True),
+     pd.Categorical(['a', 'b'], ordered=True)),
+    ([pd.Interval(1, 2), pd.Interval(3, 4)], 'interval',
+     pd.IntervalArray.from_tuples([(1, 2), (3, 4)])),
+    ([0, 1], 'Sparse[int64]', pd.SparseArray([0, 1], dtype='int64')),
+    ([1, None], 'Int16', integer_array([1, None], dtype='Int16')),
+
```

Comment:
can you also send in pd.Series / pd.Index of these types (or is this tested below)?

Reply:
do you test pass thru to ndarray? (e.g. maybe use any_numpy_dtype fixture with the dtype specified)

Reply:
It is tested below. Added a couple here as well though.

Not sure how `any_numpy_dtype` would be used here. We would need data to go with it.

Reply:
@TomAugspurger right you would need to construct a dummy array and test that its passing in thru

Reply:
Still not sure what this would look like, or what the test is for. I'm not especially worried about specific numpy types not getting through, and we do test that path.